smolyay/samples.py

```diff
@@ -0,0 +1,362 @@
+        return self._degree
+
+    @degree.setter
+    def degree(self, value):
+        self._degree = value
+        self._valid_cache = True
+
+    def _create(self):
+        r"""Generating the points
+
+        Generating the extrema of a Cheybshev polynomial of the first kind at
+        a given degree.
+        """
+        if self.degree > 0:
+            points = -numpy.cos(
+                numpy.pi * numpy.linspace(0, self.degree, self.degree + 1) / self.degree
+            )
+        else:
+            points = numpy.zeros(1)
+        self._points = points
+
+
+class NestedClenshawCurtisPointSet(NestedUnidimensionalPointSet):
+    r"""Generate nested Clenshaw Curtis points in accordance with a growth rule
+
+    The :attr:`points` for this interpolation scheme are the extrema of the
+    Chebyshev polynomials of the first kind on the domain :math:`[-1, 1]`:
+
+    .. math::
+
+        x_i^* = -\cos(\pi i/n), i = 0,...,n
+
+    For the special case :math:`n = 0`, there is only one point :math:`x_0^* = 0`.
+
+    The nested Clenshaw Curtis points come from the nested extrema of the
+    Chebyshev polynomials of the first kind :math:0, 2, 2^{k} where k is an
+    integer.
+
+    These extrema are symmetric about 0. When generating levels of points, each
+    level with contain a certain number of symmetric pairs except the first which
+    also includes 0. The number of points assigned to each level can vary.
+    The default, exponential growth adds points such that the new extrema
+    calculated from a Chebyshev polynomial 2^{k} is given its own level k-1.
+    Alternative growth rules may delay the addition of new extrema and have
+    intermediary empty levels, but the general order that points are added to
+    levels remains the same.
+
+    This class will generate the nested extrema in the order that corresponds
+    to the exponential growth. For increasing k, the unique extrema of Chebyshev
+    polynomial with degree 2^{k - 1} will be added followed by the extrema of the
+    polynomial with degree 2^{k} , then 2^{k + 1} , then 2^{k + 2} and so on.
+
+    Parameters
+    ----------
+    max_level ; int
+        the maximum level the points are used for
+    """
+
+    def __init__(self, max_level):
+        super().__init__(max_level)
+
+    @property
+    def domain(self):
+        """numpy.ndarray: Domain the sample points come from."""
+        return numpy.array([-1, 1])
+
+    def _create(self):
+        r"""Generating the points
+
+        Generating nested extrema of chebyshev polynomials of the first kind.
+        """
+        # create properties for levels
+        rule = lambda x: 1 if x == 0 else 2**x + 1
+        self._num_points = [rule(0)] + [
+            rule(i) - rule(i - 1) for i in range(1, self.max_level + 1)
+        ]
+        self._start_level = numpy.cumsum([0] + list(self._num_points))[:-1]
+        self._end_level = numpy.cumsum(list(self._num_points))
```

Comment:
Will this work instead? It seems easier to read to me
```suggestion
        self._num_points = numpy.ones(self.max_level + 1, dtype=int)
        self._num_points[1:] = [2**i - 2**(i-1) for i in range(1, self.max_level + 1)]

        self._start_level = numpy.zeros_like(self._num_points)
        self._start_level[1:] = numpy.cumsum(self._num_points)

        self._end_level = self._start_level + self._num_points
```

Reply:
No, when calculating the number of points per level, the first level is one less than it should be because 2**(i-1) is 1 when i = 1.

Reply:
I was able to make adjustments so it would work.